mkosi/machine.py

```diff
@@ -239,7 +237,7 @@ def setUp(self) -> None:
         # Replacing underscores which makes name invalid.
         # Necessary for shell otherwise racing conditions to the disk image will happen.
         test_name = self.id().split(".")[3]
-        self.machine.config.hostname = test_name.replace("_", "-")
+        object.__setattr__(self.machine.config, "hostname", test_name.replace("_", "-"))
```

Comment:
Uh, what a particularly ugly hack :)

To crawl through slightly less-hidden paths
```suggestion
        self.machine.config.__dict__["hostname"] = test_name.replace("_", "-")
```